test/include/RAJA_test-reduceloc-data.hpp

```diff
@@ -0,0 +1,42 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
```

Comment:
@rchen20 you could use the methods in RAJA_test-forall-data.hpp in the test include directory and just not use the "test_array" to keep the code smaller.

Reply:
I was thinking about that; I can if I add a remove_ref to the type of the first argument in allocateforalldata, e.g. something like remove_ref<typename T> N. The indirection arrays (workarr2D, checkarr2D) are causing this to be the wrong type.

Reply:
@rhornung67 Changed this to use forall-data.